### PR TITLE
Add Item Detail View page and Refactor ItemTable

### LINKED ISSUE #123: Item edit form
image

### LINKED ISSUE #124: Item detail view
image

### LINKED ISSUE #193: Implement <DataTable /> to Item tables
## Description
Currently Item table still uses old hard coded implementation of DataTable, we need to migrate it to use the re-usable DataTable

### PR DESCRIPTION
**Pull request by luigibesani**
Closes #193
Closes #124
Closes #123 
## Description
- Update Character Detail Page
-- Add 'Item.index' route to CharacterDetail Page to render the Item Page through Inertia.

- Refactoring ItemTable Component
-- Implement the re-usable DataTable component  into the ItemTable Component in order to move away from a hard-coded table.
-- Add 'Item.show' route to ItemTable component to render ItemDetail Page when item name is clicked.

- Add Item Detail Component
-- Add CharacterItemCreateForm for the Drawer in the ItemDetail Component
-- Add ItemDetailBox Component to display Item data

- Edit Database Seeder to create Items associated to existing Characters
- Edit ItemController to prohibit viewing item's of a character which does not belong to the current User

### CODE REVIEW

File: resources/js/Components/CharacterDetail/CharacterDetailBox.tsx [hunk 1]
```diff
@@ -79,11 +81,13 @@ const CharacterDetailBox = ({character, setIsEditDrawerOpen}: CharDetailBoxPropT
...
-                            variant='contained'
-                            startIcon={<IosShareIcon fontSize='small' />}>
-                            ITEMS
-                        </Button>
+                        <Link href={route('item.index', {character_id: character.id})}>
+                            <Button
+                                variant='contained'
+                                startIcon={<IosShareIcon fontSize='small' />}>
+                                ITEMS
+                            </Button>{' '}
```
Comment (willyyhuang) [inline resources/js/Components/CharacterDetail/CharacterDetailBox.tsx:89]: remove `{' '}` please

File: resources/js/Components/ItemDetail/ItemDetailBox.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        <Grid container rowSpacing={2}>
+            <Grid container item xs={5}>
+                <Grid item xs={4}>
+                    <Link href={route('character.show', [character.id])}>
+                        <Typography>{character.name}</Typography>
+                    </Link>
+                </Grid>
+                <Grid item xs={4}>
+                    <Typography>
+                        {' '}
```
Comment (willyyhuang) [inline resources/js/Components/ItemDetail/ItemDetailBox.tsx:40]: can you use ```{``}``` to merge these 2 lines into a single line

File: resources/js/Components/ItemDetail/ItemDetailBox.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                <Grid item xs={4} />
+            </Grid>
+            <Grid item xs={7} />
+            <Grid item xs={12} />
+        </Grid>
+        <Grid container spacing={6}>
+            <Grid item xs={6}>
+                <Img
+                    alt='avatar'
+                    src='https://static.wikia.nocookie.net/eladriells-dd/images/d/d8/Orc.jpg/revision/latest/scale-to-width-down/600?cb=20190330101155'
```
Comment (willyyhuang) [inline resources/js/Components/ItemDetail/ItemDetailBox.tsx:53]: this is a placeholder?

**Review by pascwhale — CHANGES_REQUESTED** [review]
Hey @luigibesani, everything works as intended but I do have some requests in terms of UI.

Item Detail Page
1- Can you replace the placeholder picture with a better one? The one we have for the character page is just fine.
2- Can you change how the rarity value is shown? You can look at my code for the DM Entry form from issue #211.
3- Can you align the breadcrumb component `Name`>`Item` to be on a single line? Also, can you follow the color pattern on the [mockups](https://www.figma.com/file/5uF4bHcHFTm13iMyrBzMt0/Designs-and-Mockups?node-id=593%3A24170)? The character name is a bit darker. I do appreciate the fact that you made the character name clickable!
4- Finally, this might be out of scope, but the search makes the table width change. Maybe that's something we can fix in a different issue if it's part of a custom component.

The team might also want to re-discuss whether we will allow a user to create magic items outside of (DM)Entries. I think that's all I have to say. I'll be happy to rereview the PR once the changes are done.

**Review by willyyhuang — APPROVED** [review]

**Comment by m-triassi** [comment]
@luigibesani Please let us know if you're going to address @pascwhale's comments and resolve the merge conflicts

File: resources/js/Components/Form/ItemEditForm.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                        fullWidth
+                        id='description'
+                        label='Description'
+                        name='Description'
+                        value={data.description}
+                        onChange={(e) => setData('description', e.target.value)}
+                    />
+                    {errors?.description && <ErrorText message={errors?.description} />}
+                </StyledGrid>
+                <StyledGrid item xs={12}>
```
Comment (willyyhuang) [inline resources/js/Components/Form/ItemEditForm.tsx:51]: can you use the same thing as me for rarity in `ItemForm`, rarity should be a select

File: resources/js/Components/Form/ItemEditForm.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                <Grid item xs={4} />
+                <Grid item xs={4}>
+                    <Button
+                        variant='contained'
+                        fullWidth
+                        onClick={() => {
+                            put(route('item.update', [editData.id]))
+                            if (!Object.keys(errors).length) {
+                                clearErrors()
+                                if (onCloseDrawer) {
```
Comment (willyyhuang) [inline resources/js/Components/Form/ItemEditForm.tsx:108]: this is not required, since onCloseDrawer is required

**Comment by pascwhale** [comment]
image
I tried importing a character from the `grod.csv` we've been using as an example. The Description field should avoid rendering anything below it.

Here's an example of when I manually attach an item through a DM Entry.
image
The field below it is empty. Maybe we can just put the words `No description` to provide some kind of feedback to the user.

Also, the word `DESCRPITION` is misspelled.

**Review by willyyhuang — APPROVED** [review]

**Pull request merged**